airbyte-integrations/connectors/source-github/acceptance-test-config.yml

```diff
@@ -4,6 +4,8 @@ acceptance_tests:
   spec:
     tests:
     - spec_path: "source_github/spec.json"
+      backward_compatibility_tests_config:
+        disable_for_version: "0.4.9"
```

Comment:
Why disable it? Are the changes breaking?

Reply:
Yes, CAT tests treats this change as backward incompatible

```
    def check_fake_previous_config_against_actual_spec(fake_previous_config):
        if isinstance(fake_previous_config, dict):  # Looks like hypothesis-jsonschema not only generate dict objects...
            fake_previous_config = SecretDict(fake_previous_config)
            filtered_fake_previous_config = {key: value for key, value in fake_previous_config.data.items() if not key.startswith("_")}
            try:
                jsonschema.validate(instance=filtered_fake_previous_config, schema=actual_connector_spec.connectionSpecification)
            except jsonschema.exceptions.ValidationError as err:
>               raise NonBackwardCompatibleError(err, BackwardIncompatibilityContext.SPEC)
E               connector_acceptance_test.utils.backward_compatibility.NonBackwardCompatibleError: BackwardIncompatibilityContext.SPEC - '' does not match '^([\\w.-]+/(\\*|[\\w.-]+)\\s+)*[\\w.-]+/(\\*|[\\w.-]+)$'
E               
E               Failed validating 'pattern' in schema['properties']['repository']:
E                   {'description': 'Space-delimited list of GitHub '
E                                   'organizations/repositories, e.g. `airbytehq/airbyte` '
E                                   'for single repository, `airbytehq/*` for get all '
E                                   'repositories from organization and `airbytehq/airbyte '
E                                   'airbytehq/another-repo` for multiple repositories.',
E                    'examples': ['airbytehq/airbyte airbytehq/another-repo',
E                                 'airbytehq/*',
E                                 'airbytehq/airbyte'],
E                    'order': 2,
E                    'pattern': '^([\\w.-]+/(\\*|[\\w.-]+)\\s+)*[\\w.-]+/(\\*|[\\w.-]+)$',
E                    'pattern_descriptor': 'org/repo org/another-repo org/*',
E                    'title': 'GitHub Repositories',
E                    'type': 'string'}
E               
E               On instance['repository']:
E                   ''

../../bases/connector-acceptance-test/connector_acceptance_test/utils/backward_compatibility.py:203: NonBackwardCompatibleError
```

Reply:
Do we expect users to have saved configurations that don't match the pattern? Or would those have all definitely never succeeded?

Reply:
Yes, I just check on `cloud.airbyte.com` and 34 from 526 configs failed because of a new pattern.
But they never succeeded before too (they of course passed `jsonschema` validation and failed on subsequent stages)

P.S. if you want to look at failed examples, let me know I will write them to slack.

Reply:
Got it. Thanks! So that means that although it is a schema change, the ones that would fail the validation are already failing, and the ones that were succeededing were succeeding because it already matched the new pattern. 

The backwards compatibility tests however failed because the fake previous config is generated off of the old spec(?)

Reply:
Actually, let's talk about this over slack - I guess we will see other errors for those users whose syncs are already failing (i.e. they will fail earlier) - there might be some communication to do around this?

Reply:
> the ones that would fail the validation are already failing

yes

> The backwards compatibility tests however failed because the fake previous config is generated off of the old spec(?)

1. Generated fake config = `{"repository": ""} `
2. Apply fake config to old spec - PASSED
3. Apply fake config to new spec - FAILED
4. Compatabilty broken.